### ISSUE TITLE
No data delivered with `custom_notification`

### ISSUE THREAD

**Issue by haggholm** [evollu/react-native-fcm#584]
`react-native@0.48.4`
`react-native-fcm@9.7.0`

Our app works fine with ordinary GCM notifications, but I wanted to use the `custom_notification` feature in order to run some logic on receiving background messages. However, when the app ‘wakes up’ on clicking either the library’s generated notification, or a manually created local notification via `FCM.presentLocalNotification()`, the notification that arrives through `FCM.getInitialNotification()` contains no data: only the flag `opened_from_drawer` and an empty `fcm: {}`. I’ve previously seen that when I had misconfigured the Android manifest and the intent matching the `click_action` was not associated with the main activity; but as mentioned above, it’s now working fine with regular GCM notifications.

**Comment by evollu** [comment]
can you check FCM.on('notification' callback? if app is in background, it might be possible that the notification goes there

**Comment by haggholm** [comment]
The notification callback gets the proper events when the GCM message is sent, but when the app is *opened*, there are no data. I suppose I could cache the results of the last received GCM notification, but then how do I tell what notification was clicked to open the app?

Or maybe I misunderstand?

**Comment by evollu** [comment]
did you send custom_notification together with notification payload?

**Comment by vlaraort** [comment]
Did you fixed this? I have the same problem

**Comment by haggholm** [comment]
@evollu No, I was sending only `custom_notification` for Android devices (as I believe the documentation directs).

(Apologies for late answer, I had moved onto other things and missed the notification.)

**Comment by evollu** [comment]
can you post your payload here?
